### PR TITLE
Added Descope as an OAuth2 provider

### PR DESCRIPTION
**Pull request by gaokevin1**
**What is this feature?**

Added docs for customers of ours that have asked us how to use Descope with Grafana. We wanted to make sure they can easily find these docs on both our website and Grafana's.

**Why do we need this feature?**

We've had a few Grafana customers that are migrating from their own authentication or Auth0 to use us, and want an easy guide to follow how to switch their OAuth2 providers in Grafana.

**Who is this feature for?**

Current and Future Grafana Customers who also use Descope Authentication

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Comment by gaokevin1** [comment]
@chri2547 I don't believe this needs to be part of the changelog, but I'm not able to add the label to my PR. Could you do that for me to pass the checks above?

File: docs/sources/setup-grafana/configure-security/configure-authentication/generic-oauth/index.md [hunk 1]
```diff
@@ -326,6 +326,38 @@ Payload:
...
+   auto_login = true
+   team_ids =
+   allowed_organizations =
+   name = Descope
+   client_id = <Descope Project ID>
+   client_secret = <Descope Access Key>
+   scopes = openid profile email descope.claims descope.custom_claims
+   auth_url = https://api.descope.com/oauth2/v1/authorize
+   token_url = https://api.descope.com/oauth2/v1/token
+   api_url = "https://api.descope.com/oauth2/v1/userinfo
```
Comment (IevaVasiljeva) [inline docs/sources/setup-grafana/configure-security/configure-authentication/generic-oauth/index.md:356]: nit:
```suggestion
   api_url = https://api.descope.com/oauth2/v1/userinfo
```

File: docs/sources/setup-grafana/configure-security/configure-authentication/generic-oauth/index.md [hunk 2]
```diff
@@ -326,6 +326,38 @@ Payload:
...
+
+1. Update the `[auth.generic_oauth]` section of the Grafana configuration file using the values from the **Settings** tab:
+
+   > **Note:** You can get your Client ID (Descope Project ID) under [Project Settings](https://app.descope.com/settings/project). Your Client Secret (Descope Access Key) can be generated under [Access Keys](https://app.descope.com/accesskeys).
+
+   ```bash
+   [auth.generic_oauth]
+   enabled = true
+   allow_sign_up = true
+   auto_login = true
```
Comment (IevaVasiljeva) [inline docs/sources/setup-grafana/configure-security/configure-authentication/generic-oauth/index.md:347]: nit: I'd keep this to the default value of `false`, and then the user can choose if they want to enable auto login for this integration or not.
```suggestion
   auto_login = false
```

**Review by IevaVasiljeva — APPROVED** [review]
Left some small comments, but otherwise looks good. Thank you for the contribution, we appreciate it!

**Comment by gaokevin1** [comment]
Hi @IevaVasiljeva, sorry I missed your comments. I made the changes you requested, thank you for approving my PR so quickly. If you could merge this now, that would be great!

File: docs/sources/setup-grafana/configure-security/configure-authentication/generic-oauth/index.md [hunk 3]
```diff
@@ -326,6 +326,38 @@ Payload:
...
+
+1. Create an Descope Project [here](https://app.descope.com/gettingStarted), and go through the Getting Started Wizard to configure your authentication. You can skip step if you already have Descope project set up.
+
+1. If you wish to use a flow besides `Sign Up or In`, go to the **IdP Applications** menu in the console, and select your IdP application. Then alter the **Flow Hosting URL** query parameter `?flow=sign-up-or-in` to change which flow id you wish to use.
+
+1. Click **Save**.
+
+1. Update the `[auth.generic_oauth]` section of the Grafana configuration file using the values from the **Settings** tab:
+
+   > **Note:** You can get your Client ID (Descope Project ID) under [Project Settings](https://app.descope.com/settings/project). Your Client Secret (Descope Access Key) can be generated under [Access Keys](https://app.descope.com/accesskeys).
```
Comment (lwandz13) [inline docs/sources/setup-grafana/configure-security/configure-authentication/generic-oauth/index.md:341]: {{% admonition type="note" %}}
You can get your Client ID (Descope Project ID) under [Project Settings](https://app.descope.com/settings/project). Your Client Secret (Descope Access Key) can be generated under [Access Keys](https://app.descope.com/accesskeys).
{{% /admonition %}}

File: docs/sources/setup-grafana/configure-security/configure-authentication/generic-oauth/index.md [hunk 4]
```diff
@@ -326,6 +326,38 @@ Payload:
 
 This section includes examples of setting up generic OAuth2 integration.
 
+### Set up OAuth2 with Descope
+
+To set up generic OAuth2 authentication with Descope, follow these steps:
+
+1. Create an Descope Project [here](https://app.descope.com/gettingStarted), and go through the Getting Started Wizard to configure your authentication. You can skip step if you already have Descope project set up.
```
Comment (lwandz13) [inline docs/sources/setup-grafana/configure-security/configure-authentication/generic-oauth/index.md:333]:
```suggestion
1. Create a Descope Project [here](https://app.descope.com/gettingStarted), and go through the Getting Started Wizard to configure your authentication. You can skip step if you already have Descope project set up.
```

**Review by lwandz13 — CHANGES_REQUESTED** [review]

**Review by lwandz13 — APPROVED** [review]
Thanks for your submission @gaokevin1 !!

**Comment by gaokevin1** [comment]
@lwandz13 are you going to merge this once all of the checks pass?

**Comment by lwandz13** [comment]
@gaokevin1  I need to checkout your branch to run a command, but am having issues.  I'll merge this once I can run that command.

**Pull request merged**